src/gistEncoder.js

```diff
@@ -0,0 +1,51 @@
+const fs = require("fs").promises;
+const path = require("path");
+const { OpusEncoder } = require("@discordjs/opus");
+const WaveFile = require("wavefile").WaveFile;
+
+async function encodePCM(file) {
+  const wav_file = await fs.readFile(file);
+  const wav = new WaveFile(wav_file);
+
+  // Check some of the file properties
+  console.log("Sample rate:::\t\t", wav.fmt.sampleRate);
+  console.log("Nr channels:::\t\t", wav.fmt.numChannels);
+
+  const encoder = new OpusEncoder(wav.fmt.sampleRate, wav.fmt.numChannels);
+  // encoder.setBitrate(48000); // make sure it sounds like ****
+
+  let samples = wav.getSamples(true, Int16Array);
+  console.log("Samples:::\t\t", samples.length);
+
+  let ptime = 20; // ms
+  let samples_per_frame =
+    (wav.fmt.sampleRate / (1000 / ptime)) * wav.fmt.numChannels;
+  console.log("Samples per frame:::\t", samples_per_frame);
+
+  let output_data = new Uint8Array(samples.length * 2);
+  for (let i = 0; i < samples.length - 1; i += samples_per_frame) {
+    const lastIndex = Math.min(samples.length, i + samples_per_frame);
```

Comment:
The last chunk breaks the code. In original version it was slicing from `i` to a number bigger than the samples `length`. I adapted to use the `samples.length` as max but still breaks. So I ignored it but also wonder how to fix 🤔

Reply:
your input is not divisible by frame_size. Either remove last frame as you have done here, or you zero pad it. (append zeros) to get correct length